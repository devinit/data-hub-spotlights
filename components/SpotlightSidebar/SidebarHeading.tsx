import React, { FunctionComponent } from 'react';

interface SidebarHeadingProps {
  heading?: string;
  onClick?: () => void;
}

const SidebarHeading: FunctionComponent<SidebarHeadingProps> = ({ heading, onClick }) => {
  return (
    <h2 className="spotlight__heading">
      { heading }
      <button data-testid="spotlight-menu-trigger" type="button" className="button button--minor" onClick={ onClick }>
        <i role="presentation" aria-hidden="true" className="ico ico--20 ico-arrow-down-blank"/>
      </button>
    </h2>
  );
};

export { SidebarHeading };
