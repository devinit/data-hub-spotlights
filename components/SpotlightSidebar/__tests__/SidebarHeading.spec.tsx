/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import { SidebarHeading } from '../SidebarHeading';

describe('SidebarHeading', () => {
  test('renders correctly with a heading', () => {
    const renderer = TestRenderer.create(<SidebarHeading heading="My Heading" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('has a button that responds to click events', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<SidebarHeading onClick={onClick} />);
    const button = getByTestId('spotlight-menu-trigger');
    expect(button).toBeDefined();

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalled();
  });
});
