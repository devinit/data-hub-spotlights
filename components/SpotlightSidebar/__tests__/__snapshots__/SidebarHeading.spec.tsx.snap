// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarHeading renders correctly with a heading 1`] = `
<h2
  className="spotlight__heading"
>
  My Heading
  <button
    className="button button--minor"
    data-testid="spotlight-menu-trigger"
    type="button"
  >
    <i
      aria-hidden="true"
      className="ico ico--20 ico-arrow-down-blank"
      role="presentation"
    />
  </button>
</h2>
`;
